### ISSUE TITLE
Add Status Dashboard

### ISSUE THREAD

**Issue by flowirtz** [openclimatefix/ocf-infrastructure#49]
We want to have an internal status dashboard for Nowcasting that let's us see the service health at a glance.

## Architecture
To not reinvent the wheel, we can try to use Prometheus for that.
There is [a hosted Prometheus service on AWS](https://aws.amazon.com/prometheus/) that takes care of storing and aggregating the metrics (backend). We will still need a Prometheus instance to scrape the metrics, but that could live as a docker container in our cluster.

There are helper libraries for almost all major languages, so we can use those libraries to expose metrics from our individual services. Once all our metrics are in Prometheus, we can make use of existing tooling like the Alertmanager for alerting and Grafana for visualising these metrics.

```mermaid
  graph TD;
      A[Data Source 1]-->B[Prometheus Ingester on Docker];
      E[Data Source 2]-->B[Prometheus Ingester on Docker];
      B[Prometheus Ingester on Docker]-->C[Prometheus Managed Service on AWS];
      C[Prometheus Managed Service on AWS]-->D[Grafana];
```

## Further reading
- [Blog: Using Amazon Managed Service for Prometheus to monitor EC2 environments](https://aws.amazon.com/blogs/opensource/using-amazon-managed-service-for-prometheus-to-monitor-ec2-environments/)
- [Best Practices: Metric and label naming](https://prometheus.io/docs/practices/naming/)
- [Example: Next.js Metric Exposer](https://codesandbox.io/s/next-js-api-prom-client-forked-ono9j?file=/pages/api/metrics.ts)

**Comment by peterdudfield** [comment]
Hey @flowirtz , great work with this. Do you have a website link for this deployment?

**Comment by flowirtz** [comment]
I did some work on this last week, but didn't manage to finish it.
Mainly, there is now a prometheus dashboard and an ingester in our TF `development` env.
I also started exposing prometheus metrics on our nowcasting website.

Last I checked there was some permissions issue though, for the docker crawler container (`Prometheus Ingester on Docker` in above chart) to push to the AWS-hosted service. Didn't look into that further yet.

There is also a AWS-hosted version of Grafana, that connects easily to their prometheus thing but also to CloudWatch. So once that permissions bug is resolved we should be almost there, I think.
